src/expr.cpp

```diff
@@ -2048,11 +2064,20 @@ static ConstExpr *lConstFoldBinaryIntOp(BinaryExpr::Op op, const T *v0, const T
     int count = carg0->Count();
 
     switch (op) {
-        FOLD_OP_REF(BinaryExpr::Shl, <<, TRef);
         FOLD_OP_REF(BinaryExpr::Shr, >>, TRef);
         FOLD_OP_REF(BinaryExpr::BitAnd, &, TRef);
         FOLD_OP_REF(BinaryExpr::BitXor, ^, TRef);
         FOLD_OP_REF(BinaryExpr::BitOr, |, TRef);
+
+    case BinaryExpr::Shl:
+        for (int i = 0; i < count; ++i) {
+            result[i] = (v0[i] << v1[i]);
+            if (v1[i] > (T)countLeadingZeros(v0[i])) {
```

Comment:
This is a little weird, LHS generally speaking doesn't have type ``T``. I.e. to shift ``char`` you don't need to shift by ``char``, you will shift it by ``int``.

But for the sake of this transformation this is irrelevant, as we just dealing with constant, which need to be less or equal to 64.

Nevertheless, casting to ``T`` is not needed here, as comparison operator in C requires standard integer promotion, which mean at least 32 bit integers will be compared.